app/code/Magento/ProductOverrideDataExporter/Model/Provider/Override/PriceOverrides.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\ProductOverrideDataExporter\Model\Provider\Override;
+
+use Magento\QueryXml\Model\QueryProcessor;
+use Magento\Customer\Model\Group;
+use Magento\Store\Api\Data\WebsiteInterface;
+
+class PriceOverrides
+{
+    /**
+     * @var QueryProcessor
+     */
+    private $queryProcessor;
+
+    public function __construct(
+        QueryProcessor $queryProcessor
+    )
+    {
+        $this->queryProcessor = $queryProcessor;
+    }
+
+    private function priceDiffersFromDefault(array $default, array $override): bool
+    {
+        return (
+            $default['prices']['minimumPrice']['finalPrice'] != $override['finalPrice'] ||
+            $default['prices']['minimumPrice']['regularPrice'] != $override['regularPrice']
+        );
+    }
+
+    /**
+     * Format data to reflect the message structure
+     *
+     * @param array $row
+     * @return array
+     */
+    private function format(array $row): array
+    {
+        return [
+            'productId' => $row['productId'],
+            'websiteCode' => $row['websiteCode'],
+            'customerGroupCode' => $row['customerGroupCode'],
+            'prices' => [
+                'minimumPrice' => [
+                    'finalPrice' => $row['finalPrice'],
+                    'regularPrice' => $row['regularPrice']
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @param array $values
+     * @return array
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function get(array $values): array
+    {
+        foreach ($values as $value) {
+            $queryArguments['entityIds'][] = $value['productId'];
```

Comment:
Is this the injection of `placeholders` in the query XML?

Reply:
not exactly, it rather preparation of arguments that will be passed